### ISSUE TITLE
How can I configure my own webpart to take advantage of this Chrome Addin?

### ISSUE THREAD

**Issue by mikezimm** [s-KaiNet/sp-formatter#1]
@s-KaiNet , @hugoabernier,
I really like this addin and was wondering if it's possible to build my SPFx property pane control so that the addin just finds it when it's open and applies the styling.

For instance, can I assign my property pane control the same class as the column formatting box so that it looks the same as the out of the box list formatting box?

If so, what's the minimum structure I might need to wrap my property config in to get it to just work?

![image](https://user-images.githubusercontent.com/49648086/77171294-9ffaba80-6a92-11ea-933b-4803ebd3254e.png)

**Comment by s-KaiNet** [comment]
Hi, sorry I think I didn't catch the idea. Do you mean, that you want the Chrome extension to apply the same "enhanced code view" to your property pane area?

**Comment by mikezimm** [comment]
Yes basically trick your add in to also apply the styling to any box I want in my SPFx webpart that is expecting JSON strings.

The reason is that sometimes I use a simple text box to add a JSON object for advanced properties that I don't have property pane controls for.  Sometimes that's because I want a back door to change options I may not want the average user to change.  Sometimes it's because during development, updating a web part property via the pane is much quicker and easier than building all the props and property pane controls for something that may not be needed in the final design.

Hope that makes sense.

**Comment by hugoabernier** [comment]
@mikezimm you can use the Monaco Editor property pane control I wrote for https://github.com/SharePoint/sp-dev-fx-webparts/tree/master/samples/react-enhanced-list-formatting

**Comment by mikezimm** [comment]
@hugoabernier, I think what you mentioned was that you created a property pane control that imported the Monaco library directly into your webpart.  And my reply was asking how much it increased the bundle size.

What I'm saying is design the property pane control with the right classes or "events" that the Addin recognizes so that the Addin does the work.

The benefits are that it is purely on the user side, it would hopefully be a smaller webpart because it would not need to include all the Monaco libraries, in my case most people would never need it so why burden the webpart with the code if we can let the addin do it.

Hope that makes sense.  If that's what your control already does then I mis-understood.

**Comment by s-KaiNet** [comment]
When it comes to add-in, then it's very less likely that it will work only by using the same css classes in your markup. The add-in is tightly coupled to column formatting DOM tree and uses some react hacks to synchronize data between column\view. Most likely you will see javascript errors when you try to reproduce the same DOM structure as column formatting uses.   

To your particular problem I think there two choices: 
 - create a separate Chrome addin, which applies vscode-like monaco editing experience based on a predefined selector
 - OR create a property pane control with a toggle "Enable/disable monaco view". Use ["await import"](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/dynamic-loading) feature to load heavy monaco packages only when the toggle is ON

**Comment by mikezimm** [comment]
@hugoabernier , I think you are the resident expert since you already did some work using Monaco.  

Based on the above comment, do you think you might be able modify your property pane control (or create an alternate one) that does this?  

_OR create a property pane control with a toggle "Enable/disable monaco view". Use "await import" feature to load heavy monaco packages only when the toggle is ON_

I could create a property pane toggle and text box, but getting it all working with the addin is above my skill set.

To be honest, I'm crossing my fingers hoping you see value in this alternative approach for your own use and make another great prop pane control.  But if it's not worth it I get it, I'm sure I can re-use your existing prop pane control.
